src/freenet/client/filter/HTMLFilter.java

```diff
@@ -2616,7 +2619,11 @@ void processStyle(HTMLParseContext pc) {
 			String href = getHashString(h, "href");
 			if (href != null) {
 				href = HTMLDecoder.decode(href);
-				href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				if (isIcon) {
+					href = htmlSanitizeURI(href, type, null, null, pc.cb, pc, false);
+				} else {
+					href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				}
```

Comment:
http://www.w3.org/TR/2014/REC-html5-20141028/links.html#linkTypes says there are only three values for “rel” which allow “href” to point to external resources. “stylesheet” is already handled by the code, support “link” I added with the PR, and “prefetch” is already removed by the logic involving `standardRelTypes` (see [HTMLFilter.java:2644](https://github.com/Bombe/fred/blob/1caf4622211ac5349a4ef4786df4265c4e3fb856/src/freenet/client/filter/HTMLFilter.java#L2644)).